internal/controller/management/serviceaccounts/serviceaccounts.go

```diff
@@ -0,0 +1,113 @@
+package serviceaccounts
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	rbacapi "github.com/akuity/kargo/api/rbac/v1alpha1"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+// Old annotation keys
+const (
+	OldAnnotationKeySub    = "rbac.kargo.akuity.io/sub"
+	OldAnnotationKeyEmail  = "rbac.kargo.akuity.io/email"
+	OldAnnotationKeyGroups = "rbac.kargo.akuity.io/groups"
+)
+
+// ServiceAccountReconciler Reconciles for ServiceAccounts that need annotation migration
+type ServiceAccountReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Logger *logging.Logger
+}
+
+// SetupWithManager sets up the reconciler with the manager
+func (r *ServiceAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ServiceAccount{}).
+		WithEventFilter(predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return hasOldAnnotations(e.ObjectNew.GetAnnotations())
+			},
+			CreateFunc: func(e event.CreateEvent) bool {
+				return hasOldAnnotations(e.Object.GetAnnotations())
+			},
+			GenericFunc: func(e event.GenericEvent) bool {
+				return hasOldAnnotations(e.Object.GetAnnotations())
+			},
+		}).
+		Complete(r)
+}
+
+// Reconcile handles the reconciliation logic for migrating annotations
+func (r *ServiceAccountReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	sa := &corev1.ServiceAccount{}
+	err := r.Get(ctx, req.NamespacedName, sa)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		r.Logger.Error(err, "Failed to get ServiceAccount")
+		return ctrl.Result{}, err
```

Comment:
Similar to [here](https://github.com/akuity/kargo/pull/2579/files#r1777293104)

```suggestion
		return ctrl.Result{}, fmt.Errorf("failed to get ServiceAccount: %w", err)
```